### PR TITLE
💄 #554 2/3: add assentFormAcknowledged to invite form UI

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI
- Add assent form acknowledged field to invite form
- Update schema with new field

### Types
- Add field refinement for assent form acknowledged
- Update tests

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: packages/types/src/services/consentUi/utils/guardianFields.ts [hunk 1]
```diff
@@ -184,3 +187,34 @@ export const hasRequiredRegistrationPhoneNumberForGuardianStatus = (
...
+export const hasRequiredAssentFormAcknowledgementForConsentGroup = (
+	props: ClinicianInviteFormAssentFormAcknowledgedField,
+	ctx: z.RefinementCtx,
+) => {
+	const { assentFormAcknowledged, consentGroup } = props;
+	if (consentGroup === GUARDIAN_CONSENT_OF_MINOR_INCLUDING_ASSENT) {
+		if (assentFormAcknowledged !== true) {
+			addZodCustomError(
+				ctx,
+				'assentFormAcknowledged',
```
Comment (anncatton) [inline packages/types/src/services/consentUi/utils/guardianFields.ts:207]: not a blocker to this PR, but once your other zod error enum ticket is in we can update these error messages

**Review by anncatton — COMMENTED** [review]
lgtm, just a couple typos! I'll leave it up to you if you want to get #535 in first so you can add in the error enums, or update in a separate pr.

File: apps/consent-ui/src/components/views/Invite/ClinicianInviteForm/index.tsx [hunk 2]
```diff
@@ -130,15 +132,14 @@ const ClinicianInviteFormComponent = ({
 		handleEnableSubmit(isValid, recaptchaToken);
 	}, [getRecaptchaToken, isValid]);
 
-	// watch consentGroup value & show/hide guardian info fields if participant is a minor.
-	// guardian fields register on mount, in their input component.
+	// watch consentGroup value & show guardian info fields if participant is a minor.
+	// show the asset form acknowledgement field, if the invitee is a guardian with assent.
+	// guardian & asset form fields register on mount, in their input components.
```
[the remark this answers is not in the document]
Comment (samrichca) [reply apps/consent-ui/src/components/views/Invite/ClinicianInviteForm/index.tsx:137]: thanks, assent stopped looking like a word 😂

File: packages/types/src/services/consentUi/utils/ZodCustomErrors.ts [hunk 3]
```diff
@@ -25,6 +25,8 @@ const ZOD_CUSTOM_ERRORS = [
 	'guardianHasParticipantInfo',
 	'guardianHasParticipantPhoneNumber',
 	'guardianInfoMissing',
+	'missingAssentFormAcknowledgementForConsentGroup',
```
Comment (anncatton) [inline packages/types/src/services/consentUi/utils/ZodCustomErrors.ts:28]: 👍

**Review by anncatton — APPROVED** [review]
Just realized we can use the enum values in some of the tests, i.e. `expect(message)toBe(error)`, but don't worry about that for now. Looks good!

**Pull request merged**